docs/DeveloperGuide.md

```diff
@@ -2,14 +2,372 @@
+     Throws `GpaNullException` if there are no modules with countable grades.
+    
+   - **`groupModulesBySemester()`**  
+     Groups modules by their semester and returns a map where each key is a semester, and each value is a list of modules in that semester.
+    
+   #### Error Handling
+    
+   Uses exceptions to handle errors, such as when trying to access or modify modules that don't exist.
+    
+5. **Getting module details from Json File (JsonManager Class):**
+    
+   #### Overview
+    
+   The `JsonManager` class is designed to manage and interact with module information stored in a JSON format. It provides functionalities for checking the existence of modules, retrieving module information such as Modular Credits (MCs), description, and title from a JSON file.
+    
+   #### Constructor
+    
+   - `JsonManager()`: Initializes a new instance of the `JsonManager` by loading the module information from a JSON file located at `/moduleInfo.json`.
+   
+   #### Methods
+    
+   ##### Module Existence
+    
+   - **`moduleExist(String moduleCode)`**: Checks if a module with the specified code exists in the JSON data.
+     - **Parameters**: `String moduleCode` - The code of the module to check for existence.
+     - **Returns**: `boolean` - `true` if the module exists, `false` otherwise.
+    
+   ##### Module Information Retrieval
+    
+   - **`getModuleInfo(String moduleCode)`**: Retrieves detailed information about a module, including its Modular Credits, description, and title, based on the module code.
+     - **Parameters**: `String moduleCode` - The code of the module for which information is to be retrieved.
+     - **Note**: This method updates the internal state of the `JsonManager` object with the retrieved module information.
+    
+   ##### Information Accessors
+    
+   - **`getModuleDescription()`**: Returns the description of the last module queried.
+     - **Returns**: `String` - The description of the module.
+   
+   - **`getModuleMC()`**: Returns the Modular Credits of the last module queried.
+     - **Returns**: `int` - The Modular Credits of the module.
+    
+   - **`getModuleTitle()`**: Returns the title of the last module queried.
+     - **Returns**: `String` - The title of the module.
+    
+   #### Error Handling
+    
+   - The constructor throws a `RuntimeException` if the JSON file containing module information cannot be found or accessed, ensuring that the application is aware of missing or inaccessible module data.
+    
+   #### Usage
+    
+   ```java
+   JsonManager jsonManager = new JsonManager();
+   if (jsonManager.moduleExist("CS1010")) {
+       jsonManager.getModuleInfo("CS1010");
+       System.out.println("Module Title: " + jsonManager.getModuleTitle());
+       System.out.println("Module Description: " + jsonManager.getModuleDescription());
+       System.out.println("Module MC: " + jsonManager.getModuleMC());
+   }
+   ```
+6. **Viewing modules left to graduate**
```

Comment:
Is it possible to include sequence diagrams for each of the command class for visualisation?